docs/rfcs/0001-integration-with-dune.md

```diff
@@ -0,0 +1,169 @@
+- Start Date: 2021-05-31
+- RFC PR: https://github.com/melange-re/melange/pull/200
+
+# Summary
+
+To allow Melange users to leverage Dune `library` and `executable` stanzas as
+it is currently possible with any other OCaml project.
+
+# Basic example
+
+Dune would allow to use Melange as another cross-compiler, e.g.
+
+```dune
+(context (default (targets (melange))))
+```
+
+Then, users would be able to define `library` and `executable` stanzas that
+would be built by calling Melange binary `ocamlc`. To define the output format
```

Comment:
I'm not a big fan of calling the binary `ocamlc`, we're using in hacking `ocamlc` because it's already setup by dune, but it makes things really confusing.

Reply:
Hm I'm confused. If we don't use `ocamlc`, then what's the point of defining melange through cross-compilation?

Reply:
Because the dependencies are different, the advantage of the cross compiler workflow is important because you want two dependency set, one for melange and another one for native.

Of course there is some other ways to separate the deps, but this is a deeper problem and the current solution by dune is having multiple findlib.conf.d